### PR TITLE
util/docker: fix network parsing

### PR DESCRIPTION
**Pull request by JulienBalestra**
### What does this PR do?

Fix the parsing of the network IPs in the docker util package.

I added some tests to cover the public function `DefaultGateway`.

### CODE REVIEW

**Comment by xvello** [comment]
This is fix for a bug impacting infra agent & process agent in non-standard network configurations, let's add a reno `fixes` entry

**Review by xvello — DISMISSED** [review]
Logic LGTM, waiting on @sunhay to review for Burito
This impacts all process-agent versions if the cluster uses this GW IP type.

File: pkg/util/docker/network.go [hunk 1]
```diff
@@ -120,11 +120,19 @@ func findDockerNetworks(containerID string, pid int, container types.Container)
 		if fields[1] == "00000000" {
 			continue
 		}
-		dest, _ := strconv.ParseInt(fields[1], 16, 32)
-		mask, _ := strconv.ParseInt(fields[7], 16, 32)
-		for net, gw := range dockerGateways {
+		dest, err := strconv.ParseUint(fields[1], 16, 32)
```
Comment (sunhay) [inline pkg/util/docker/network.go]: Shouldn't this be 64 bits instead of 32 now?

**Review by sunhay — DISMISSED** [review]
Just one question on my side

File: pkg/util/docker/network.go (hunk 1 above)
Comment (truthbk) [reply pkg/util/docker/network.go]: IPv4 should always fit in 32 bits, and `/proc/net/route` is IPv4 only, being `/proc/net/ipv6_route` the source for ipv6 routing tables.

**Pull request merged**